test/test_real_pebble.py

```diff
@@ -153,70 +155,69 @@ def test_exec_wait(self):
         process = self.client.exec(['true'])
         process.wait()
 
-        with self.assertRaises(pebble.ExecError) as cm:
+        with pytest.raises(pebble.ExecError) as excinfo:
             process = self.client.exec(['/bin/sh', '-c', 'exit 42'])
             process.wait()
-        self.assertEqual(cm.exception.exit_code, 42)
+        assert excinfo.value.exit_code == 42
 
     def test_exec_wait_output(self):
         process = self.client.exec(['/bin/sh', '-c', 'echo OUT; echo ERR >&2'])
         out, err = process.wait_output()
-        self.assertEqual(out, 'OUT\n')
-        self.assertEqual(err, 'ERR\n')
+        assert out == 'OUT\n'
+        assert err == 'ERR\n'
 
         process = self.client.exec(['/bin/sh', '-c', 'echo OUT; echo ERR >&2'], encoding=None)
         out, err = process.wait_output()
-        self.assertEqual(out, b'OUT\n')
-        self.assertEqual(err, b'ERR\n')
+        assert out == b'OUT\n'
+        assert err == b'ERR\n'
 
-        with self.assertRaises(pebble.ExecError) as cm:
+        with pytest.raises(pebble.ExecError) as excinfo:
             process = self.client.exec(['/bin/sh', '-c', 'echo OUT; echo ERR >&2; exit 42'])
             process.wait_output()
-        exc = typing.cast(pebble.ExecError[str], cm.exception)
-        self.assertEqual(exc.exit_code, 42)
-        self.assertEqual(exc.stdout, 'OUT\n')
-        self.assertEqual(exc.stderr, 'ERR\n')
+        assert excinfo.value.exit_code == 42
+        assert excinfo.value.stdout == 'OUT\n'  # type: ignore
```

Comment:
Can we keep the cast to avoid the ignores?

Reply:
Casting works as expected as shown in the CI real pebble test. Resolving now.